cli/commands/traffic_step.go

```diff
@@ -114,24 +114,29 @@ func (cmd *TrafficStepCommand) Action(ctx *cli.Context) error {
 }
 
 // StepTraffic .
-func (cmd *TrafficStepCommand) StepTraffic(serviceName string, amount float32) error {
+func (cmd *TrafficStepCommand) StepTraffic(serviceName string, amount float64) error {
 
-	routes, err := cmd.controller.ServiceRoutes(serviceName)
+	filter := &api.RuleFilter{
+		Destinations: []string{serviceName},
+	}
+
+	routes, err := cmd.controller.ListRoutes(filter)
 	if err != nil {
 		return err
 	}
 
-	if len(routes.Rules) == 0 {
+	routingRules := routes.Services[serviceName]
+	if len(routingRules) == 0 {
 		fmt.Fprintf(cmd.ctx.App.Writer, "%s: %q\n\n", common.ErrNotRulesFoundForService.Error(), serviceName)
 		return nil
 	}
 
-	if len(routes.Rules) > 1 || len(routes.Rules[0].Route.Backends) != 2 || routes.Rules[0].Route.Backends[0].Weight == routes.Rules[0].Route.Backends[1].Weight {
+	if len(routingRules) > 1 || len(routingRules[0].Route.Backends) != 2 || routingRules[0].Route.Backends[0].Weight == routingRules[0].Route.Backends[1].Weight {
```

Comment:
I think it would make sense to create a function for this if we use this "if" in more than 2 places. Since that's not the case at this time, I don't think is necessary.